### ISSUE TITLE
EXCEPTION CAUGHT BY WIDGETS LIBRARY

### ISSUE THREAD

**Issue by ZhipengMei** [jonasbark/flutter_stripe_payment#2]
Hi, I ran the example code on iOS and the following message is shown. 

```

flutter: ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
flutter: The following assertion was thrown attaching to the render tree:
flutter: type 'Future<dynamic>' is not a subtype of type 'Future<String>' where
flutter:   Future is from dart:async
flutter:   Future is from dart:async
flutter:   String is from dart:core
flutter:
flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
flutter: more information in this error message to help you determine and fix the underlying cause.
flutter: In either case, please report this assertion by filing a bug on GitHub:
flutter:   https://github.com/flutter/flutter/issues/new
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #0      StripeSource.addSource (package:stripe_payment/stripe_payment.dart:11:51)
flutter: #1      _MyAppState.initState (file:///Users/adrian/Documents/development/flutter/First_Flutter_Project/stripe_api/lib/main.dart:21:18)
flutter: #2      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:3743:58)
flutter: #3      ComponentElement.mount (package:flutter/src/widgets/framework.dart:3609:5)
flutter: #4      Element.inflateWidget (package:flutter/src/widgets/framework.dart:2899:14)
flutter: #5      Element.updateChild (package:flutter/src/widgets/framework.dart:2702:12)
flutter: #6      RenderObjectToWidgetElement._rebuild (package:flutter/src/widgets/binding.dart:857:16)
flutter: #7      RenderObjectToWidgetElement.mount (package:flutter/src/widgets/binding.dart:828:5)
flutter: #8      RenderObjectToWidgetAdapter.attachToRenderTree.<anonymous closure> (package:flutter/src/widgets/binding.dart:774:17)
flutter: #9      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2214:19)
flutter: #10     RenderObjectToWidgetAdapter.attachToRenderTree (package:flutter/src/widgets/binding.dart:773:13)
flutter: #11     _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding&WidgetsBinding.attachRootWidget (package:flutter/src/widgets/binding.dart:662:7)
flutter: #12     runApp (package:flutter/src/widgets/binding.dart:704:7)
flutter: #13     main (file:///Users/adrian/Documents/development/flutter/First_Flutter_Project/stripe_api/lib/main.dart:6:3)
flutter: #14     _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:279:19)
flutter: #15     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:165:12)
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
```

Also another question, is the billing address support coming along in the next update or no?

**Comment by jonasbark** [comment]
I'm pretty sure the fault is in your code. How are you calling `addSource`?

No, billing address support (of which I'm not even sure what you mean) is not in the scope of this library.

**Comment by jonasbark** [comment]
Please provide more details. How did you implement the library, e.g. how do you call `addSource`?

**Comment by AndrewPetrovics** [comment]
Well, now it's just giving me an unhandled exception error, but I'm implementing just like in the example; straight copy and paste and I'm currently on the master channel of Flutter using version 0.0.4.

`[VERBOSE-2:dart_error.cc(16)] Unhandled exception:`

**Comment by jonasbark** [comment]
please add the output of `flutter doctor -v`
Are you using dart2?

**Comment by AndrewPetrovics** [comment]
Here's `flutter doctor -v` output:

`Andrews-MacBook-Pro-4:Pointz andrew$ flutter doctor -v
[✓] Flutter (Channel master, v0.4.4, on Mac OS X 10.12.6 16G1212, locale en-US)
    • Flutter version 0.4.4 at /Users/Andrew/dev/flutter
    • Framework revision f9bb4289e9 (4 days ago), 2018-05-11 21:44:54 -0700
    • Engine revision 06afdfe54e
    • Dart version 2.0.0-dev.54.0.flutter-46ab040e58

[!] Android toolchain - develop for Android devices (Android SDK 26.0.1)
    • Android SDK at /Users/andrew/Android
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-26, build-tools 26.0.1
    • ANDROID_HOME = /Users/andrew/Android
    • Java binary at: /Applications/Android Studio 2.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)
    ✗ Android license status unknown.

[✓] iOS toolchain - develop for iOS devices (Xcode 9.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 9.1, Build version 9B55
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.0

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio 2.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)`

Yes, I'm assuming I am using dart2 based on the output (I'm new to Flutter / dart).

I should also add that the Stripe screen does indeed pop up and show. It just doesn't resolve the "then" promise.

**Comment by jonasbark** [comment]
Fixed with 0.0.6
